scripts/pylib/twister/twisterlib/handlers.py

```diff
@@ -1019,8 +1020,8 @@ def handle(self, harness):
                 self.returncode = proc.returncode
             # Need to wait for harness to finish processing
             # output from QEMU. Otherwise it might miss some
-            # error messages.
-            self.thread.join(0)
+            # messages.
+            self.thread.join(max(thread_max_time - time.time(), 0))
```

Comment:
without much refactoring here, I think it is better just to let the qemu `_thread()` do its own timeout logic capturing output and closeing fifo-s.

Reply:
according to the python thread document, it says if the thread timeout, it will keeps running, so this this case, the qemu instance thread will be kept running.

Reply:
> according to the python thread document, it says if the thread timeout, it will keeps running, so this this case, the qemu instance thread will be kept running.

right, it is what's needed here

Reply:
> @golowanow do you add the kill thread?

No, if the log monitoring thread is still running we just wait a bit more here for what remains of the timeout.
The thread might run even longer though with its `timeout_extended` logic, but it is the next problem to fix.

> without much refactoring here, I think it is better just to let the qemu `_thread()` do its own timeout logic capturing output and closing fifo-s.